### DISCUSSION TITLE
Add buildDependencies separate from devDependencies

### DISCUSSION THREAD

**Discussion by WViiras** [npm/rfcs#500] [RRFC]
We have a project that is built on microfrontends with multiple repositories for each frontend. In order for them be available for developers and support autocomplete between the different projects, we've added them to devDependencies list, but they're not needed during the build itself, so there's no real need to have them be downloaded for a build either. 
The same goes for other @Types/linting/testing/etc packages where they're needed for the development but not needed for building the final product.
It would be nice if there would be a separate block for build only dependencies that would be separate from dev dependencies.

**Comment by jedwards1211** [comment]
I can imagine variations on what groups of deps people want to install... Maybe it would be best to support defining groups of deps in package.json that you can refer to from the install command.  Though it would probably be pretty easy for custom code to read groups from package.json and install only the deps in a given group
